# Titanic Disaster

![Titanic realna fotka](https://raw.githubusercontent.com/Masterx-AI/Project_Titanic_Survival_Prediction_/main/titanic.jpg)

originalny dataset z Kaggle.com je dostupny tu: https://www.kaggle.com/competitions/titanic

## Data Reorganization

1. chceme spojit dataset `test.csv` s datasetom `train.csv`
2. pozriet sa na duplikaty
3. rozdelit stlpec `Name` na `firstname` a `lastname`

### 1. Spojenie datasetov

In [1]:
from pathlib import Path
import pandas as pd

path = Path('data/titanic')

In [10]:
df_train = pd.read_csv(path / 'train.csv', index_col='PassengerId')
df_test = pd.read_csv(path / 'test.csv', index_col='PassengerId')
df_survival = pd.read_csv(path / 'gender_submission.csv', index_col='PassengerId')

zobrazime si rozmery jedneho aj druheho dataset-u pomocou vlastnosti `.shape`:

In [12]:
df_train.shape, df_test.shape

((891, 11), (418, 10))

ked sa pozrieme na ne blizsie, tak vieme, ze:

* dataset `train` ma `891` riadkov a `11` stlpcov
* dataset `test` ma `418` riadkov a `10` stlpcov

po spojeni by sme mali v sucte `891 + 418 = 1309` riadkov. ale mame problem so stlpcami - su rozdielne. nemozeme spojit dva datasety, ktore maju rozdielny pocet stlpcov (iba ak by sme niektore vynechali alebo by se vytvorili prazdny stlpec).

pozrime sa teda na to, aky stlpec nam chyba v dataset `test`:

In [14]:
df_train.columns, df_test.columns

(Index(['Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket',
        'Fare', 'Cabin', 'Embarked'],
       dtype='object'),
 Index(['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare',
        'Cabin', 'Embarked'],
       dtype='object'))

a je to jasne. chyba nam hned prvy stlpec `Survived`. hodnoty tohto stlpca sa totiz nachadzaju v datasete `gender_submission.csv`. 

spojenie teda bude vyzerat takto:

* najprv spojime datasety `gender_submission.csv` a `test.csv` na zaklade stlpca `PassengerId`
* potom spojime datasety `train.csv` s vyslednym datasetom, ktory vznikol spojenim `gender_submission.csv` a `test.csv` rovnako na zaklade stlpca `PassengerId`

In [15]:
# spojenie datasetov gender_submission.csv a test.csv


In [18]:
# spojenie dataset train.csv a vysledneho datasetu, ktory vznikol v predchadzajucom kroku
